interfaces/adfsuite/ams.py

```diff
@@ -585,14 +629,14 @@ def get_band_structure(self, bands=None, unit: str = "hartree", only_high_symmet
 
         return x, complete_spinup_data, complete_spindown_data, labels, fermi_energy
 
-    def get_engine_results(self, engine: str = None):
+    def get_engine_results(self, engine: str = None) -> Dict:
```

Comment:
Since we are touching these lines already: `None` is not a `str`, so the correct type-hint here would be `Optional[str]`.

Lots of other places where this is wrong.

Reply:
Fixed... `mypy` is my new friend